subsys/usb/device_next/class/usbd_midi.c

```diff
@@ -0,0 +1,752 @@
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_cs_out_ep,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_in_ep_fs,         \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_cs_in_ep,         \
+		NULL,                                                                    \
+	};                                                                               \
+	static const struct usb_desc_header *usbd_midi_desc_array_hs_##inst[] = {        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.iad,                    \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if0_std,                \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if0_cs,                 \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_std,              \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_ms_header,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_out_ep_hs,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_cs_out_ep,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_in_ep_hs,         \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_cs_in_ep,         \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_std,              \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_ms_header,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_out_ep_hs,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_cs_out_ep,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_in_ep_hs,         \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_cs_in_ep,         \
+		NULL,                                                                    \
+	};
+
+#define USBD_MIDI_DEFINE_DEVICE(inst)                                           \
+	USBD_MIDI_VALIDATE_INSTANCE(inst)                                       \
+	USBD_MIDI_DEFINE_DESCRIPTORS(inst);                                     \
+	USBD_DEFINE_CLASS(midi_##inst, &usbd_midi_class_api,                    \
+			  (void *)DEVICE_DT_GET(DT_DRV_INST(inst)), NULL);      \
+	static const struct usbd_midi_config usbd_midi_config_##inst = {        \
+		.desc = &usbd_midi_desc_##inst,                                 \
+		.fs_descs = usbd_midi_desc_array_fs_##inst,                     \
+		.hs_descs = usbd_midi_desc_array_hs_##inst,                     \
+	};                                                                      \
+	static struct usbd_midi_data usbd_midi_data_##inst = {                  \
+		.class_data = &midi_##inst,                                     \
+		.altsetting = ALT_USB_MIDI_1,                                   \
+	};                                                                      \
+	DEVICE_DT_INST_DEFINE(inst, usbd_midi_preinit, NULL,                    \
+			      &usbd_midi_data_##inst, &usbd_midi_config_##inst, \
+			      POST_KERNEL, CONFIG_SERIAL_INIT_PRIORITY, NULL);
```

Comment:
`CONFIG_SERIAL_INIT_PRIORITY`? you probably wanted to add a specific init priority symbol? or just use `KERNEL_INIT_PRIORITY_DEVICE` I guess